packages/medusa/src/repositories/product.ts

```diff
@@ -198,7 +206,7 @@ export const ProductRepository = dataSource.getRepository(Product).extend({
     }
 
     if (salesChannelId) {
-      const joinMethod = options_.relations.sales_channel_id
+      const joinMethod = options_.relations.sales_channels
```

Comment:
@adrien2p It doesn't make sense why this is `sales_channel_id` instead of `sales_channels`, can you confirm if this is correct? 

Or should i do `options_.where.sales_channel_id`?

Reply:
Confirming that the change made is indeed correct. I've applied the same fix to categories down as well.